hack/goarch.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this script attempts to print GOARCH for the host
+
+# if we have go, just ask go!
+if which go >/dev/null 2>&1; then
+    go env GOARCH
+    exit $?
+fi
+
+# fall back to uname
+if which uname >/dev/null 2>&1; then
+  case "$(uname -m)" in
+    x86_64)
+      echo "amd64"
+      exit 0
+    ;;
+    arm*)
+      if [ "$(getconf LONG_BIT)" == "64" ]; then
```

Comment:
this will probably need to be more elaborate, but arm users can also just `go build` and for the release we'll cross build anyhow instead of detecting the host.